src/improvements/AddressRegistry.sol

```diff
@@ -413,31 +413,27 @@ contract AddressRegistry is IAddressRegistry, Test {
         view
         returns (address)
     {
-        try IFetcher(faultDisputeGame).anchorStateRegistry() returns (address anchorStateRegistryProxy) {
-            return anchorStateRegistryProxy;
-        } catch {
-            return IFetcher(permissionedDisputeGame).anchorStateRegistry();
-        }
+        return IFetcher(permissionedDisputeGame).anchorStateRegistry();
     }
 
     function getDelayedWETHProxy(address faultDisputeGame, address permissionedDisputeGame)
         internal
         view
         returns (address)
     {
-        try IFetcher(faultDisputeGame).weth() returns (address delayedWethProxy) {
-            return delayedWethProxy;
-        } catch {
-            return IFetcher(permissionedDisputeGame).weth();
-        }
+        (bool ok, bytes memory data) =
+            address(faultDisputeGame).staticcall(abi.encodeWithSelector(IFetcher.weth.selector));
+        if (ok && data.length == 32) return abi.decode(data, (address));
+
+        return IFetcher(permissionedDisputeGame).weth();
```

Comment:
This code block is inaccurate since there is one delayed WETH proxy per game type. So the FDG has it's own delayed weth, and the PDG has its own delayed weth. I would change this method to take a single dispute game input, then call it once or twice (i.e. once for the PDG, once for the FDG if it's present)

Reply:
Does this mean there should be two 'DelayedWETHProxy' addresses saved to the AddressRegistry is a chain has both FDG and PDG? Or should we only save one? If FDG is present then save it's DelayedWETHProxy otherwise default to saving the PDG DelayedWETHProxy.

Reply:
> there should be two 'DelayedWETHProxy' addresses saved to the AddressRegistry is a chain has both FDG and PDG

Exactly. Actually, I would probably just stop the lookup at the dispute game addresses, save those, and a task can fetch the delayed WETH addresses if needed. No need to store the WETH addresses. This will make it more maintainable, because eventually we will add a third game type (asterisc)

Reply:
Would another approach be to look at the current game type on the OptimismPortalProxy via `resepectedGameType()`? https://etherscan.io/address/0xbEb5Fc579115071764c7423A4f12eDde41f106Ed#readProxyContract

Once you have that, you can decide which delayed weth proxy to read (GameType = 0, read FDG, GameType = 1, read PDG).

Reply:
Not in the generic case. For example if you are changing the implementation of the delayedWETH contract, you need to check the new impls for both DelayedWETHs

Reply:
We don't know what a task will do, so we can't make assumptions about which delayedWETH it will want

Reply:
https://github.com/ethereum-optimism/superchain-ops/pull/582/commits/a695a7bb747b372f465e763892df68edc7987569